internal/handlers/common/projection.go

```diff
@@ -38,20 +34,8 @@ func isProjectionInclusion(projection *types.Document) (inclusion bool, err erro
 		switch v := v.(type) {
 		case *types.Document:
 			for _, projectionType := range v.Keys() {
-				supportedProjectionTypes := []string{"$elemMatch", "$slice"}
-				if !slices.Contains(supportedProjectionTypes, projectionType) {
-					err = lazyerrors.Errorf("projection of %s is not supported", projectionType)
-					return
-				}
-
-				switch projectionType {
-				case "$elemMatch":
-					inclusion = true
-				case "$slice":
-					inclusion = false
-				default:
-					panic(projectionType + " not supported")
-				}
+				err = lazyerrors.Errorf("projection of %s is not supported", projectionType)
```

Comment:
I see this `lazyerror` was already in the code before. I'm not sure why we don't return `NewCommandErrorMsgWithArgument(ErrNotImplemented, ...))` in this case though.